AlgorithmContext.cpp

```diff
@@ -1,13 +1,20 @@
 #include "AlgorithmContext.h"
 #include <chrono>
 
-AlgorithmContext::AlgorithmContext(Algorithm *algo_ = nullptr) : algo(algo_) {}
+AlgorithmContext::AlgorithmContext(Algorithm *algo_,

```

Comment:
why removing default argument?

Reply:
Then I have to provide a default value for the new argument `io_` too and I thought there's no use to default construct it here as we always construct `AlgorithmContext` for a specific IO instance that has already read a graph file. But I can provide two default arguments here if you prefer it this way.

Reply:
nah, we just need to remember to put nullptr when creating Context, AO